src/diffusers/models/transformers/lumina_nextdit2d.py

```diff
@@ -0,0 +1,507 @@
+
+        self.adaLN_modulation = nn.Sequential(
+            nn.SiLU(),
+            nn.Linear(
+                min(dim, 1024),
+                4 * dim,
+                bias=True,
+            ),
+        )
+
+        self.attn_encoder_hidden_states_norm = RMSNorm(
+            cross_attention_dim, eps=norm_eps, elementwise_affine=norm_elementwise_affine
+        )
+
+    def forward(
+        self,
+        hidden_states: torch.Tensor,
+        attention_mask: torch.Tensor,
+        image_rotary_emb: torch.Tensor,
+        encoder_hidden_states: torch.Tensor,
+        encoder_mask: torch.Tensor,
+        adaln_input: Optional[torch.Tensor] = None,
+        cross_attention_kwargs: Dict[str, Any] = None,
+    ):
+        """
+        Perform a forward pass through the LuminaNextDiTBlock.
+
+        Args:
+            hidden_states (torch.Tensor): Input tensor.
+            image_rotary_emb (torch.Tensor): Precomputed cosine and sine frequencies.
+        Returns:
+            torch.Tensor: Output tensor after applying attention and
+                feedforward layers.
+        """
+        residual = hidden_states
+
+        if adaln_input is not None:
+            scale_msa, gate_msa, scale_mlp, gate_mlp = self.adaLN_modulation(adaln_input).chunk(4, dim=1)
+
+            # Self-attention
+            hidden_states = modulate(self.attn_norm1(hidden_states), scale_msa)
+            self_attn_output = self.attn(
+                hidden_states=hidden_states,
+                encoder_hidden_states=hidden_states,
+                attention_mask=attention_mask,
+                query_rotary_emb=image_rotary_emb,
+                key_rotary_emb=image_rotary_emb,
+                **cross_attention_kwargs,
+            )
+
+            # Cross-attention
+            norm_encoder_hidden_states = self.attn_encoder_hidden_states_norm(encoder_hidden_states)
+            cross_attn_output = self.cross_attn(
+                hidden_states=hidden_states,
+                encoder_hidden_states=norm_encoder_hidden_states,
+                attention_mask=encoder_mask,
+                query_rotary_emb=image_rotary_emb,
+                key_rotary_emb=None,
+                **cross_attention_kwargs,
+            )
+            cross_attn_output = cross_attn_output * self.gate.tanh().view(1, 1, -1, 1)
+            mixed_attn_output = self_attn_output + cross_attn_output
+            mixed_attn_output = mixed_attn_output.flatten(-2)
+            # linear proj
+            hidden_states = self.cross_attn.to_out[0](mixed_attn_output)
+
+            hidden_states = residual + gate_msa.unsqueeze(1).tanh() * self.attn_norm2(hidden_states)
+
+            mlp_output = self.feed_forward(
+                modulate(self.ffn_norm1(hidden_states), scale_mlp),
+            )
+            hidden_states = hidden_states + gate_mlp.unsqueeze(1).tanh() * self.ffn_norm2(mlp_output)
+        else:
```

Comment:
I don't think we need the `else` part because the `adaln_input` is always passed

Reply:
done

Reply:
done